lib/library.rb

```diff
@@ -0,0 +1,68 @@
+class Library
+
+  attr_reader :name, :books, :authors, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @book_checkout_count = {}
```

Comment:
good job tracking this in a hash, but I probably wouldn't expose it with an attr_reader otherwise other developers could do other things with our hash that maybe we don't want to allow.